package/src/contexts/messageInputContext/MessageInputContext.tsx

```diff
@@ -1002,32 +1002,53 @@ export const MessageInputProvider = <
   }) => {
     const id = generateRandomId();
     const mimeType = lookup(file.name);
+
+    const blockedFile = blockedFiles?.some((fileExtensionType: string) =>
+      file.name?.includes(fileExtensionType),
+    );
+    const blockedMime = blockedMimeFiles?.some((mimeType: string) => file.name?.includes(mimeType));
+
     const newFile = {
       file: { ...file, type: mimeType || file?.type },
       id,
-      state: FileState.UPLOADING,
+      state: blockedFile || blockedMime ? FileState.NOT_SUPPORTED : FileState.UPLOADING,
     };
+
     await Promise.all([
       setFileUploads((prevFileUploads) => prevFileUploads.concat([newFile])),
       setNumberOfUploads((prevNumberOfUploads) => prevNumberOfUploads + 1),
     ]);
 
-    uploadFile({ newFile });
+    if (!blockedFile) {
+      uploadFile({ newFile });
+    }
   };
 
   const uploadNewImage = async (image: Partial<Asset>) => {
     const id = generateRandomId();
+
+    const blockedMime = blockedMimeImages?.some((mimeType: string) =>
+      image.uri?.includes(mimeType),
+    );
+
+    const blockedImage = blockedImages?.some((imageExtensionType: string) =>
+      image.uri?.includes(imageExtensionType),
+    );
+
     const newImage = {
       file: image,
       id,
-      state: FileState.UPLOADING,
+      state: blockedImage || blockedMime ? FileState.NOT_SUPPORTED : FileState.UPLOADING,
```

Comment:
Same as above, would be easier to read if you extract it to a variable

Reply:
✅